.github/workflows/desktop-release.yml

```
@@ -510,3 +1009,46 @@ jobs:
+          [ "${MODE}" = "auto" ] || { echo "::error::mode is '${MODE}' on an archdisc push; it should be 'auto'"; bad=1; }
+          [ "${PUBLISH_OUTCOME}" = "success" ] || { echo "::error::the publish step did not run (outcome '${PUBLISH_OUTCOME}')"; bad=1; }
+          [ -n "${PUBLISH_STATE:-}" ] || { echo "::error::the publish step produced no state output"; bad=1; }
+          [ "${ACCEPT_OUTCOME}" = "success" ] || { echo "::error::the acceptance step did not run (outcome '${ACCEPT_OUTCOME}')"; bad=1; }
+          if [ "$bad" -ne 0 ]; then
+            echo "::error::this build went green on archdisc and did not publish a visible" \
+                 "release. That is the exact failure this workflow exists to prevent, so it" \
+                 "is a RED build rather than a quiet one."
+            exit 1
+          fi
+          echo "publish path ran: ${PUBLISH_STATE}, acceptance ${ACCEPT_OUTCOME}."
+
```

YAMLlint (1.37.1)

[error] too many blank lines (1 > 0)

(empty-lines)